### PR TITLE
Molecule migration

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
Addition of molecule class to the migration branch. Should be merged after segment class.

### CODE REVIEW

File: include/votca/xtp/segment.h [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    const double    &getU_cC_nN(int state);
+    const double    &getU_nC_nN(int state);
+    const double    &getU_cN_cC(int state);
+    const double    &getU_xX_nN(int state);
+    const double    &getU_nX_nN(int state);
+    const double    &getU_xN_xX(int state);
+    double           getSiteEnergy(int state);
+
+    double           getEMpoles(int state);
+    void             setEMpoles(int state, double energy);
```
Comment (JensWehner) [inline include/votca/xtp/segment.h]: why does segment have this functionality and what does it do?

File: src/libxtp/molecule.cc [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    throw runtime_error( string("Not implemented") ); 
+}
+
+/// Returns number of atoms in the molecule
+int Molecule::NumberOfAtoms() {
+    return _atoms.size(); 
+}
+
+/// Read atom types and coordinates from a file
+void Molecule::ReadXYZ(string filename)
```
Comment (JensWehner) [inline src/libxtp/molecule.cc]: another reader

File: src/libxtp/molecule.cc [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        //vec position(x,y,z);
+        //pAtom->setPos( position );                       
+        //AddAtom( pAtom );  
+
+    }      
+    in.close();        
+}
+
+/// Writes a PDB file
+void Molecule::WritePDB( FILE *out ) {
```
Comment (JensWehner) [inline src/libxtp/molecule.cc]: another writer, holy shit. I know it is not your fault but wow

File: include/votca/xtp/segment.h (hunk 1 above)
Comment (JoshuaSBrown) [reply include/votca/xtp/segment.h]: No idea. We will look at it during the refactoring stage.

**Pull request merged**